Yasgui/packages/yasqe/src/index.ts

```diff
@@ -272,12 +273,16 @@ export class Yasqe extends CodeMirror {
     return this.pageSize;
   }
 
-  public setIsExplainPlanQuery(isExplainPlanQuery: boolean): void {
-    this.isExplainPlanQuery = isExplainPlanQuery;
+  public setExplainPlanQueryType(explainType?: EXPLAIN_PLAN_TYPE): void {
+    this.explainPlanQueryType = explainType;
   }
 
-  public getIsExplainPlanQuery(): boolean | undefined {
-    return this.isExplainPlanQuery;
+  public getExplainPlanQueryType(): EXPLAIN_PLAN_TYPE | undefined {
+    return this.explainPlanQueryType;
+  }
+
+  public isExplainPlanQuery(): boolean {
+    return !!this.explainPlanQueryType;
```

Comment:
Here the idea is to find out if the query type is `explain` and not the concrete type. Right?

Reply:
Yes. You are right.